src/pages/send/amount/amount.ts

```diff
@@ -466,7 +466,35 @@ export class AmountPage extends WalletTabsChild {
     return result;
   }
 
-  public finish(): void {
+  public checkAmount() {
+    if (!this.wallet || this.requestingAmount) {
+      this.finish();
+      return;
+    }
+
+    let unit = this.availableUnits[this.unitIndex];
+    let _amount = this.evaluate(this.format(this.expression));
+    let amount = unit.isFiat
+      ? (this.fromFiat(_amount) * this.unitToSatoshi).toFixed(0)
+      : (_amount * this.unitToSatoshi).toFixed(0);
+    let maxAmountAvailable = this.wallet.status.availableBalanceSat;
+
+    if (amount > maxAmountAvailable) {
+      const insufficientFundsInfoSheet = this.actionSheetProvider.createInfoSheet(
+        'insufficient-funds',
+        { amount: _amount, coin: unit.shortName }
+      );
+      insufficientFundsInfoSheet.present();
+      insufficientFundsInfoSheet.onDidDismiss(option => {
+        if (!option) {
+          this.sendMax();
```

Comment:
This function shouldn't be called before `finish()`, because there is a `zone.run` inside. The variable `expression` could be unpredictable

Reply:
You're right. This line was removed. Thanks!